core/src/labels/labelCollider.cpp

```diff
@@ -251,6 +265,8 @@ void LabelCollider::process(TileID _tileID, float _tileInverseScale, float _tile
         }
     }
 
+    filterRepeatGroups(repeatPos, m_labels.size()-1);
```

Comment:
This checks for the last range of groups and in general it converges quite quickly, but have you considered doing all the filtering at once at the end?